src/extension/utils.js

```diff
@@ -326,8 +362,20 @@ export function getProjectMatches(configs, tabUrl) {
     return checkHost === prodHost // production host
       || checkHost === previewHost // custom inner
       || checkHost === liveHost // custom outer
-      || checkHost.split('.hlx.')[0].endsWith(`${repo}--${owner}`) // inner or outer
-      || queryDiscoveryCache(tabUrl).find((e) => e.owner === owner && e.repo === repo);
+      || checkHost.split('.hlx.')[0].endsWith(`${repo}--${owner}`); // inner or outer
+  });
+  (await queryUrlCache(tabUrl)).forEach((e) => {
+    // add non-duplicate matches from url cache
+    const filteredByUrlCache = configs.filter(({ owner, repo }) => {
+      if (e.owner === owner && e.repo === repo) {
+        // avoid duplicates
+        if (!matches.find((m) => m.owner === owner && m.repo === repo)) {
+          return true;
+        }
+      }
+      return false;
+    });
+    matches = matches.concat(filteredByUrlCache);
   });
```

Comment:
I'm sure there's a more elegant way to do this, but here we scan the URL cache for matches with the current owner/repo combo and add them to the array of matches, avoiding duplicates.